code/modules/atmospherics/machinery/datum_pipeline.dm

```diff
@@ -179,10 +179,21 @@
 /datum/pipeline/proc/temporarily_store_air()
 	//Update individual gas_mixtures by volume ratio
 
+	var/sealed_volume = 0
+
+	for(var/obj/machinery/atmospherics/pipe/member as anything in members)
+		if(!member.sealed)
+			continue
+		sealed_volume += member.volume
+
 	for(var/obj/machinery/atmospherics/pipe/member in members)
+		if(member.sealed)
+			continue
 		member.air_temporary = new
-		member.air_temporary.volume = member.volume
-		member.air_temporary.copy_from(air, member.volume / air.volume)
+		member.air_temporary.volume = member.volume + sealed_volume
+		member.air_temporary.copy_from(air, (member.volume + sealed_volume) / air.volume)
+		if(sealed_volume > 0)
+			sealed_volume = 0
```

Comment:
I don't like how you're using sealed volume here.
Instead of adding all of the sealed gasses into one net, I'd much prefer just deceasing the total air volume by the sealed volume. That way things equalize normally.

Though now that I think about it that means if you stick seals on half a pipeline and cut at the center, the half just doesn't get any air, which is dumb, and also something this implementation runs into.

Maybe have unwrenching a sealed pipe manually remove it from the pipeline's members list? it's a bit fragile, but it's less inconsistent

Reply:
> Maybe have unwrenching a sealed pipe manually remove it from the pipeline's members list? it's a bit fragile, but it's less inconsistent

This may be the better way to do it, i too was icky about this, but didn't know another way to do it, i'll try this other thing